packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1501,25 +1486,15 @@ class StripeHelper {
       },
     } = invoice;
 
-    if (
-      !charge ||
-      !charge.payment_method_details ||
-      !charge.payment_method_details.card
-    ) {
-      throw error.internalValidationError(
-        'extractSubscriptionUpdateReactivationDetailsForEmail',
-        invoice,
-        new Error(
-          `Expected card payment method details for invoice: ${invoice.id}`
-        )
-      );
+    let lastFour: string | null = null;
+    let cardType: string | null = null;
+    if (charge?.payment_method_details?.card) {
+      ({
+        brand: cardType,
+        last4: lastFour,
+      } = charge.payment_method_details.card);
```

Comment:
nit: these few lines are identical to those above, so perhaps wrap them into a function.